api/Service/EventService.cs

```diff
@@ -48,16 +49,31 @@ public async Task<Event> CreateEvent(EventDto eventDto)
 				Tags = eventDto.Tags,
 				BackgroundId = await _pictureService.CreatePictureByContent(eventDto.BackgroundImage)
 			};
+
+			var updateCategory = await _categoryService.GetCategory(eventDto.CategoryId);
+			updateCategory.HasEvents = true;
+			await _categoryService.PutCategory(eventDto.CategoryId, updateCategory);
+			
 			return newEvent;
 		}
 
 		public async Task<bool> DeleteEvent(int eventID)
 		{
 			var myEvent = await _context.Events.FindAsync(eventID);
+
 			if (myEvent == null)
 			{
 				return false;
 			}
+
+			var eventsWithSameCategory = await _context.Events.Where(ev => ev.CategoryId == myEvent.CategoryId).ToListAsync();
+			if (eventsWithSameCategory.Count == 1)
```

Comment:
You can use SingleOrDefaultAsync() instead of Where and check if it's not null.

Reply:
Is it okay if I use FirstOrDefaultAsync instead of SingleOrDefaultAsync() ?